src/main/java/com/project/service/UserServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+
+    private final UserRepository userRepository;
+    private BCryptPasswordEncoder bcryptEncoder;
+    private final PasswordTokenRepository passwordTokenRepository;
+
+
+    @Autowired
+    public UserServiceImpl(UserRepository userRepository, BCryptPasswordEncoder bcryptEncoder, PasswordTokenRepository passwordTokenRepository) {
+        this.userRepository = userRepository;
+        this.bcryptEncoder = bcryptEncoder;
+        this.passwordTokenRepository = passwordTokenRepository;
+    }
+
+    @Override
+    public List<UserResponse> getAll() {
+        List<User> users = userRepository.findAll();
+        return users.stream().map(UserResponse::new).collect(Collectors.toList());
+    }
+
+    @Override
+    public User getById(final Long id) throws UserNotFoundException {
+        return userRepository.findById(id).orElseThrow(() -> new UserNotFoundException(id));
+    }
+
+    @Override
+    public User getByEmail(String email) throws UsernameNotFoundException {
+        return userRepository.findByEmail(email).orElseThrow(() -> new UsernameNotFoundException(email));
+    }
+
+    @Override
+    public User addUser(UserRequest userRequest) throws UserException {
+        if (userRepository.findByEmail(userRequest.getEmail()).isPresent()) {
+            throw new UserEmailExistsException(userRequest.getEmail());
+        }
+
+        userRequest.setPassword(bcryptEncoder.encode(userRequest.getPassword()));
+
+        User user = new User(userRequest);
+
+        List<String> roles = new ArrayList<>();
+        roles.add(UserRole.USER.name());
+        user.setRoles(roles);
+
+        return userRepository.save(user);
+    }
+
+    @Override
+    public UserResponse updateUser(UserRequest user, Long id) throws UserException {
+        Optional<User> existingUser = userRepository.findById(id);
+
+        if (!existingUser.isPresent()) {
+            throw new UserNotFoundException(id);
+        }
+
+        if (userRepository.findByEmail(user.getEmail()).isPresent()
+                && !userRepository.findByEmail(user.getEmail()).get().getId().equals(id)) {
+            throw new UserEmailExistsException(user.getEmail());
+        }
+
+        User response = existingUser
+                .map(existing -> {
+                    existing.setEmail(user.getEmail());
+                    existing.setPassword(user.getPassword());
+                    existing.setFullName(user.getFullName());
+                    return userRepository.save(existing);
+                }).get();
+        return new UserResponse(response);
+    }
+
+    @Override
+    public void deleteUser(long id) throws UserNotFoundException {
+        if (!userRepository.existsById(id)) {
+            throw new UserNotFoundException(id);
+        }
+        userRepository.deleteById(id);
+    }
+
+    @Override
+    public void createPasswordResetTokenForUser(User user, String token) {
+        PasswordResetToken myToken = new PasswordResetToken(token, user);
```

Comment:
Can it be just something like `resetToken` and not `myToken`?

Reply:
renamed variable from `myToken` to `resetToken`